### DISCUSSION TITLE
IBUS telemetry - GPS and other sensors not shown on Transmitter

### DISCUSSION THREAD

**Discussion by alccrl** [OpenI6X/opentx#324] [Q&A] [answered]
HI,

(openi6x-1.8.0))

Plying around with betaflight 4.3.2 and a SkyStars H7 HD FC, with compass, barometer and gps, I noticed at a point (IBUS telemetry on UART) two sensors "GPS" appearing in telemetry discovery (page 10 on open-i6x).
After deletion of all sensors and trying again discovery,  I was not anymore able to see these GPS line in telemetry list (page 10).

I tried several times to get again a telemetry list, also looking at betaflight configuration, but I got only basic RSSI, BAT, RPM, there is no complete list of telemetry data as documented in all IBUS TM forums, github or the like. I get only a subset of them...
Receiver ia6b is flashed with RSSI on channel 8, fw found in the online pages about modding FlySky receivers (Cleric-k),
and is configure as PPM/IBUS. RSSI works (on ch8).

Is there anyway to configure transmitter fs-ia6x to get all declared data (including gps data) in IBUS tm, or there are bugs/limitations about ?

Any idea,

Alfredo

**Comment by Ultraexpert** [comment]
I have also had this where I get altitude and gps but only one time

**Comment by ajjjjjjjj** [comment]
In INAV you have to configure IBUS telemetry to report all sensors with `ibus_telemetry_type` parameter.
https://github.com/iNavFlight/inav/blob/master/docs/Telemetry.md#cli-command

There should be something similar in Betaflight.

**Reply by Ultraexpert** [reply]
Thanks

**Comment by alccrl** [comment]
Thanks again,

I have been looking at Telemetry.md several times, but I never focused that that list was a cli parameter list, not a telemetry list ,
very funny , indeed ...!!!

Ps:
Betaflight behaves differently, Telemetry.md is different.

Alfredo

**Comment by alccrl** [comment]
I succeded in discovering gps sensors with inav-5.1, 
it took me time because I was not able to configure a new inav target for fc Skystars H7 HD that I was working on.
After a few difficulties, I have been able to compile a working target for this FC.

Now,
I can play with IBUS telemetry types.... After a few trials, it seems that type 6 works and suffices my needs.
Netherless,
it is hard to match names with exact meaning, and gps coordinates are shown with low precision :

I get (on transmitter ) :
45.05
7.68 
instead of (on OSD) :
45.0518279
7.68015410

With 0.01 precision (sensor editing on open-i6x does not permit higher precision) values are useless for lost drone-finding.

Do I miss some configuration option ?

Is gps high precision (part) shown in other sensors, which ones, if any  ?
Moreover, inav 5.1 sources .c .h (telemetry files) show a lot of options in numbering and ordering, so it is not clear which order applies in telemetry sensors after discovery on transmitter side.
Few sensors seem to be useless (strange values, difficolt to be mapped on something physical).

Any explanation for this ?

Thanks in advance if someone have  a rational.

Alfredo

**Reply by ajjjjjjjj** [reply]
This is in INAV documentation regarding IBUS telemetry GPS precision:

> (...)
12.Second part of Lattitude (Rpm type), for example 5678 (-12.3456789 N).
13.Second part of Longitude (Rpm type), for example 6789 (-123.4567891 E).
14.First part of Lattitude (Voltage type), for example -12.45 (-12.3456789 N).
15.First part of Longitude (Voltage type), for example -123.45 (-123.4567890 E).
(...)

If you want better precision on the radio screen try `ibus_telemetry_type=4` or higher:
> 4.This same as 3, but support 4 byte sensors. (fix_updater_03_16_21_33_1 from https://github.com/qba667/FlySkyI6/tree/master/release): sensor 7 is 4byte ALT, 12 is PRESURE or PITOT_SPEED if avaliable, 13 is GPS_STATUS, 14 is 4byte GPS_LON, 15 is 4byte GPS_LAT. This required a receiver with new firmware that support SNR, RSSI and long frames

Reference: https://github.com/iNavFlight/inav/blob/master/docs/Telemetry.md

**Reply by alccrl** [reply]
Thanks,

I found type 6 suitable for me, but on transmitter all sensors seem to be in different order than what declared in inav/betaflight source code and telemetry.md.
I had also a look at open-i6x source code,  more differences there, ... so I guess that a mapping of all sensor cases using telemetry.md scheme and all firmwares available would be a very difficult task ....

Alfredo

**Comment by alccrl** [comment, marked as answer]
I succeded also in getting gps ibus telemetry with Betaflight 4.3.2,
but I had to change from AUTO to 115800 baud rate for gps and setting in cli :
set ibus_sensor = 2,3,4,5,6,7,20,21,22,126,128,129,130,131,132.

I had to search through forums to reach this result.

Alfredo

**Comment by alccrl** [comment]
Hi,

Yet more success on getting ibus gps telemetry ...
I obtained all 16 sensors working with ibus tm type 1. !!!

Netherless ..., they came in total messy order, I had to rebuild and change names and units for all of them, helping myself with inav configurator and osd data in order to understand where they were coming from.

Alfredo
